pyscraper/new_hansard.py

```diff
@@ -0,0 +1,1442 @@
+        if found_member is None:
+            if member.get('PimsId') == 0:
+                found_member = {
+                    'person_id': 'unknown',
+                    'name': u''.join(member.xpath('.//text()'))
+                }
+
+        return found_member
+
+    def parse_newdebate(self, tag):
+        time = tag.xpath('.//ns:hs_time', namespaces=self.ns_map)
+        if len(time):
+            self.parse_time(time[0])
+
+        heading = tag.xpath('.//ns:hs_DebateHeading', namespaces=self.ns_map)
+        debate_type = tag.xpath('.//ns:hs_DebateType', namespaces=self.ns_map)
+        if len(heading):
+            if len(debate_type):
+                text = self.get_single_line_text_from_element(debate_type[0])
+                self.parse_major(heading[0], extra_text=text)
+            else:
+                self.parse_major(heading[0])
+        else:
+            sys.stderr.write('newdebate with no heading', namespaces=self.ns_map)
+            return
+
+        #procedure = tag.xpath('.//ns:hs_Procedure', namespaces=self.ns_map)
+        #if len(procedure) == 1:
+        #    self.handle_para(procedure[0])
+
+        want_member = tag.get('BusinessType') in ('Question', 'GeneralDebate')
+
+        member = None
+        member_tags = tag.xpath('.//ns:Member', namespaces=self.ns_map)
+        if len(member_tags):
+            if want_member:
+                member = self.parse_member(member_tags[0])
+            else:
+                tabledby_tags = tag.xpath('.//ns:hs_TabledBy', namespaces=self.ns_map)
+                self.parse_para_with_member(tabledby_tags[0], None, css_class='italic', strip_member=False)
+
+        questions = tag.xpath('.//ns:hs_Question', namespaces=self.ns_map)
+        for question in questions:
+            self.parse_para_with_member(question, member if want_member else None)
+
+    def parse_procedure(self, procedure):
+        tag = etree.Element('p')
+        text = self.get_single_line_text_from_element(procedure)
+        if len(text) == 0:
+            return
+
+        tag.set('pid', self.get_pid())
+        tag.set('class', 'italic')
+        self.current_speech_part = self.current_speech_part + 1
+        tag.text = text
+
+        if self.current_speech is not None:
+            self.current_speech.append(tag)
+
+    def parse_amendment_heading(self, heading):
+        self.new_speech(None, heading.get('url'))
+        self.parse_para_with_member(heading, None)
+
+    def parse_tabledby(self, tabledby):
+        self.parse_para_with_member(
+            tabledby,
+            None,
+            strip_member=False,
+            css_class='italic',
+            pwmotiontext='unrecognized'
+        )
+
+    def parse_amendment(self, amendment):
+        self.parse_para_with_member(
+            amendment,
+            None,
+            css_class='italic',
+            pwmotiontext='unrecognized'
+        )
+
+    def parse_clause_heading(self, heading):
```

Comment:
In yesterday's debates, hs_8Clause contained the "New Clause 1", "Clause 5" headings that precede the actual heading title; they're currently appearing as the last paragraph of the previous section.